### DISCUSSION TITLE
Conditional Config not working (@wagmi/cli)

### DISCUSSION THREAD

**Discussion by linghuccc** [wevm/wagmi#3335] [Q&A] [answered]
wagmi.config.ts:
import { defineConfig } from '@wagmi/cli'
import { foundry, react } from '@wagmi/cli/plugins'
import * as chains from 'wagmi/chains'

export default defineConfig(() => {
	if (process.env.NEXT_PUBLIC_TESTNET === 'true') {
		return {
			out: 'src/generated_dev.ts',
...
		}
	} else {
		return {
			// production specific config
			out: 'src/generated.ts',
...
		}
	}
})

.env.local:
NEXT_PUBLIC_TESTNET=true

But when I run "npx wagmi generate", I got src/generated.ts instead of src/generated_dev.ts. 

I used "console.log" to print out "process.env.NEXT_PUBLIC_TESTNET" and it is "undefined". Then I tried "source .env.local" before running "npx wagmi generate". "process.env.NEXT_PUBLIC_TESTNET" is still "undefined" and the file generated is still src/generated.ts.

Please advise. Thank you.

**Comment by tmm** [comment, marked as answer]
Check out https://wagmi.sh/cli/configuration/configuring-cli#environment-variables

**Reply by linghuccc** [reply]
Thank you so much. It worked!
